### PR TITLE
fix rename permission model registry 

### PR DESCRIPTION
**Pull request by MohamedKHALILRouissi**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/MohamedKHALILRouissi/mlflow/pull/14139?quickstart=1)

#### Install mlflow from this PR

```
# Use `%sh` to run this command on Databricks
OPTIONS=$(if pip freeze | grep -q 'mlflow @ git+https://github.com/mlflow/mlflow.git'; then echo '--force-reinstall --no-deps'; fi)
pip install $OPTIONS git+https://github.com/mlflow/mlflow.git@refs/pull/14139/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 14139
```

</p>
</details>

### Related Issues/PRs

 #14113 

### What changes are proposed in this pull request?
Update model registry name permission when the user updates the registry name through the API or UI, ensuring users can maintain permissions after renaming.

### How is this PR tested?

- [x] Existing unit/integration tests 
- [ ] New unit/integration tests
- [X] Manual tests

multiple update ,i assigned different user to the model registry test-t-1 then renamed it to test-t-3 
image

### Does this PR require documentation update?

- [X ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [X] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="rename-permission-fix"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes.

<details>
<summary>What is a minor/patch release?</summary>

- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

@serena-ruan

### CODE REVIEW

**Comment by serena-ruan** [comment]
Could you add unit test here https://github.com/mlflow/mlflow/blob/master/tests/server/auth/test_sqlalchemy_store.py?

**Comment by MohamedKHALILRouissi** [comment]
@serena-ruan Hello, please let me know if there's anything else to add. I’d be happy to do it. Thanks!

File: tests/server/auth/test_sqlalchemy_store.py [hunk 1]
```diff
@@ -424,3 +425,29 @@ def test_delete_registered_model_permission(store):
     ) as exception_context:
         store.get_registered_model_permission(name1, username1)
     assert exception_context.value.error_code == ErrorCode.Name(RESOURCE_DOES_NOT_EXIST)
+
+
+def test_rename_registered_model_permission(store):
+    # create 2 users and create 2 permission for the model registry with the same name
+    model_registry_name = random_str()
```
Comment (serena-ruan) [inline tests/server/auth/test_sqlalchemy_store.py:432]:
```suggestion
    model_name = random_str()
```
It's `registered model's name`, not model registry

File: tests/server/auth/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -424,3 +425,29 @@ def test_delete_registered_model_permission(store):
...
+    password1 = random_str()
+    _user_maker(store, username1, password1)
+    _rmp_maker(store, model_registry_name, username1, MANAGE.name)
+
+    username2 = random_str()
+    password2 = random_str()
+    _user_maker(store, username2, password2)
+    _rmp_maker(store, model_registry_name, username2, MANAGE.name)
+
+    model_registry_new_name = random_str()
```
Comment (serena-ruan) [inline tests/server/auth/test_sqlalchemy_store.py:443]:
```suggestion
    new_name = random_str()
```

File: tests/server/auth/test_sqlalchemy_store.py [hunk 3]
```diff
@@ -424,3 +425,29 @@ def test_delete_registered_model_permission(store):
...
+
+    store.rename_registered_model_permissions(model_registry_name, model_registry_new_name)
+
+    # get permission by model registry new name and all user must have the same new name
+    perm_user_1 = store.get_registered_model_permission(model_registry_new_name, username1)
```
Comment (serena-ruan) [inline tests/server/auth/test_sqlalchemy_store.py:448]: Let's check the permission is MANAGE after renaming as well

**Review by serena-ruan — COMMENTED** [review]
Mostly LGTM! Let's update the test :D

File: tests/server/auth/test_sqlalchemy_store.py (hunk 3 above)
Comment (MohamedKHALILRouissi) [reply tests/server/auth/test_sqlalchemy_store.py]: @serena-ruan done and thanks again

Comment (MohamedKHALILRouissi) [reply tests/server/auth/test_sqlalchemy_store.py]: i added 2 type of permission to check them MANAGE(user1) , READ(user2)

**Comment by serena-ruan** [comment]
@mlflow-automation autoformat

**Comment by serena-ruan** [comment]
Could you fix the format?

**Comment by MohamedKHALILRouissi** [comment]
@serena-ruan  hello can you tell me exactly which format triggering the issue

**Comment by MohamedKHALILRouissi** [comment]
@serena-ruan  done

**Review by serena-ruan — APPROVED** [review]

**Pull request merged**